charts/linkerd2/values.yaml

```diff
@@ -2,14 +2,101 @@
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
 
-clusterDomain: &cluster_domain cluster.local
-enableH2Upgrade: true
-imagePullPolicy: &image_pull_policy IfNotPresent
+# Values that are passed along to sub-charts
+global:
+  clusterDomain: &cluster_domain cluster.local
+  imagePullPolicy: &image_pull_policy IfNotPresent
+
+  # control plane version. See Proxy section for proxy version
+  linkerdVersion: &linkerd_version edge-19.12.3
+
+  namespace: linkerd
+
+  # identity configuration
+  identity:
```

Comment:
Before fixing the docs, One concern I have is to not move `identity` to global. Only `identity.trustAnchorsPEM`, `identity.trustDomain` is needed for us. Nothing from `identity.issuer`. 

So I did some changes to move `identity.issuer` out of global, and renamed `trustAnchorsPEM`, `trustDomain` to `global.identityTrustAnchorsPEM`,`global.identityTrustDomain` (i.e out of Identity and to global). This required changes in `install.go`, etc. I took the path of using a new struct there. 

The changes can be seen in this new branch here https://github.com/Pothulapati/linkerd2/commit/0351c9b774c4b14ed5d5b7b136c101516e6e8c61 . I can push that commit here, if its acceptable and then fix the docs. 



@zaharidichev @alpeb

Reply:
This fragments things a bit but on the other hand make it quite DRY. One thing I have to say is I am not sure that `completeIdentity` is the right name for the struct. Maybe IdentitiyWithAnchorsAndTrustDomain (...not sure that sounds much better). I do not have any objections against the approach. In situations like these however the limitation of the go type system shine through :D

Reply:
@Pothulapati Agreed about the variables refactoring. As for the naming, I don't have a great suggestion. Not a big deal IMO, given it's a private helper struct.

Reply:
Updated using the helper struct.